src/App.js

```diff
@@ -1,16 +1,51 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
 import chatMessages from './data/messages.json';
 
 const App = () => {
+  const [entries, setEntries] = useState(chatMessages);
+
+  const startLikeCount = entries.filter((entry) => entry.liked).length;
+
+  const [likeCount, setLikeCount] = useState(startLikeCount);
+
+  const updateMessageData = (updatedMessage) => {
+    const messages = entries.map((message) => {
+      if (message.id === updatedMessage.id) {
+        if (updatedMessage.liked === true && message.liked === false) {
+          setLikeCount(likeCount + 1);
+        } else if (updatedMessage.liked === false && message.liked === true) {
+          setLikeCount(likeCount - 1);
```

Comment:
As mentioned before, this is where our state for the like count starts to cause a bit of inefficiency. Notice how as we go through each message in the messages list, we update the state via the changeState function. If we however had a standalone function that counted the likes once the messages have been updated, that would drastically increase the efficiency here!